api/mappers/dynamodb.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, List, TypeAlias, TypeVar, Union
+
+from pydantic import BaseModel
+
+from mappers.engine import (
+    map_environment_api_key_to_engine,
+    map_environment_to_engine,
+    map_identity_to_engine,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    from flag_engine.identities.models import IdentityModel
+
+    from environments.identities.models import Identity
+    from environments.models import Environment, EnvironmentAPIKey
+
+
+__all__ = (
+    "map_engine_identity_to_identity_document",
+    "map_environment_api_key_to_environment_api_key_document",
+    "map_environment_to_environment_document",
+    "map_identity_to_identity_document",
+)
+
+DocumentValue: TypeAlias = Union[Dict[str, "DocumentValue"], str, bool, None, Decimal]
+Document: TypeAlias = Dict[str, DocumentValue]
+
+
+def map_environment_to_environment_document(
+    environment: "Environment",
+) -> Document:
+    return {
+        field_name: _map_value_to_document_value(value)
+        for field_name, value in map_environment_to_engine(
```

Comment:
I'm not too sure how this is supposed to work. As far as I can tell, `map_environment_to_engine` returns an `EnvironmentModel` instance which is a dataclass (and hence not iterable).

Reply:
In the new engine version, models are subclasses of Pydantic's `BaseModel`, which support this way of iterating over fields and values.

Reply:
Ah, yes, of course. Nice, that's cool!